aya-obj/src/obj.rs

```diff
@@ -711,10 +719,29 @@ impl Object {
         Ok(())
     }
 
-    fn parse_program(&self, section: &Section) -> Result<(Program, Function), ParseError> {
-        let prog_sec = ProgramSection::from_str(section.name)?;
-        let name = prog_sec.name().to_owned();
+    fn parse_programs(&mut self, section: &Section) -> Result<(), ParseError> {
+        let program_section = ProgramSection::from_str(section.name)?;
+        for symbol_index in self.symbols_by_section.get(&section.index).unwrap().iter() {
```

Comment:
I don't think we should unwrap here? It's valid for a section to not have any
symbols in the symbol table